manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java

```diff
@@ -214,6 +233,30 @@ private void configurePublishingTask(Configuration config, Project project, Mani
             } else {
                 logger.warn("No publishing plugin was configured for '{}'!", evaluatedProject.getName());
             }
+
+            // If signing is enabled, automatically set it to not required.
+            Object sign = evaluatedProject.getExtensions().findByName("signing");
+            if (sign != null) {
+                SigningExtension signingExtension = ((SigningExtension) sign);
+                boolean required;
+                try {
+                    // In a decorated class it is prefixed/suffixed by '__'.
+                    Object o = FieldUtils.readDeclaredField(signingExtension, "__required__", true);
+                    if (o instanceof Boolean) {
+                        required = (Boolean) o;
+                    } else {
+                        // TODO: Not sure if this can happen?
+                        logger.debug("Retrieve signing type of {}", o);
+                        required = true;
+                    }
+                } catch (IllegalAccessException e) {
+                    throw new ManipulationUncheckedException(e);
+                }
+                if (required) {
+                    logger.warn("Signing was detected as enabled - disabling.");
+                    signingExtension.setRequired(false);
+                }
+            }
```

Comment:
This block may make removing the signing plugin redundant - and we could perhaps consider disabling that by default then ?

Reply:
I think it is better to do it in code if we can rather than relying on text modification.

Reply:
Ok, thanks, so this leads onto two further points:

* I found https://github.com/gradle/gradle/blob/master/subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java#L113 so I think this exists in >= Grade 4.10
  *  I think the signing type is always a Boolean but I'm not sure

* We currently have `pluginRemoval=ALL` as a shortcut. I think we should keep that but then that means we need a suitable extra shortcut e.g. 'recommended' (`REC`) or 'standard' (`STD`) etc. Can you suggest another good shortcut name and abbreviation to add ? Obviously this will also need a documentation update.

Reply:
1. Yes. https://github.com/gradle/gradle/blob/v4.10.3/subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java#L115
2. Maybe `DEFAULT`?

Reply:
>     2. Maybe `DEFAULT`?

While I quite like default, one contextual problem is that it is not the default - plugins are not removed by default (whether they _should_ be is another discussion!) This is why I am leaning towards 'recommended'.

Reply:
I meant default list of plugins. I mean, the option wouldn't appear unless you specified that option to begin with?

Reply:
In any case, naming is not critical IMO.